src/dmd/backend/blockopt.d

```diff
@@ -300,20 +336,20 @@ void blocklist_free(block **pb)
 
 void block_optimizer_free(block *b)
 {
-    vec_free(b->Bdom);
-    vec_free(b->Binrd);
-    vec_free(b->Boutrd);
-    vec_free(b->Binlv);
-    vec_free(b->Boutlv);
-    vec_free(b->Bin);
-    vec_free(b->Bout);
-    vec_free(b->Bgen);
-    vec_free(b->Bkill);
-    vec_free(b->Bout2);
-    vec_free(b->Bgen2);
-    vec_free(b->Bkill2);
-
-    memset(&b->_BLU,0,sizeof(b->_BLU));
+    vec_free(b.Bdom);
+    vec_free(b.Binrd);
+    vec_free(b.Boutrd);
+    vec_free(b.Binlv);
+    vec_free(b.Boutlv);
+    vec_free(b.Bin);
+    vec_free(b.Bout);
+    vec_free(b.Bgen);
+    vec_free(b.Bkill);
+    vec_free(b.Bout2);
+    vec_free(b.Bgen2);
+    vec_free(b.Bkill2);
+
+    memset(&b,0,b.sizeof);
```

Comment:
I'm not sure whether this is the correct translation.

Reply:
It isn't. It's supposed to zero the pointers to the vectors that were just freed. An easy way to do this is to write a nested function:
```
static void vfree(ref vec_t* v) { vec_free(v); v = null; }
```
and call that instead of vec_free.